plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/XStatefulIncrementalBuilder.java

```diff
@@ -109,10 +109,13 @@ public XBuildResult launch() {
 			Set<URI> remainingURIs = new LinkedHashSet<>(oldIndex.getAllURIs()); // note: creating a copy!
 
 			XIndexer.XIndexResult result = indexer.computeAndIndexDeletedAndChanged(request, context);
-			operationCanceledManager.checkCanceled(request.getCancelIndicator());
-
 			ResourceDescriptionsData newIndex = result.getNewIndex();
 			List<Delta> deltasToBeProcessed = result.getResourceDeltas();
+			List<Delta> deltasFromExternal = indexer.computeAndIndexAffected(newIndex, remainingURIs,
```

Comment:
The indexer does not have a chance to cancel this operation?

Reply:
Do you mean that the indexer should call `operationCanceledManager.checkCanceled`?